install.sh

```diff
@@ -3,123 +3,404 @@
+	printf "$_IMPORTANT \"sudo\" will be used when necessary. \n"
+	_SUDO="$(type -fp sudo)"
+	if [ -z $_SUDO ]; then
+		printf "\"sudo\" was not found on the system. Please log in as root \n"
+		printf "before running the installer, or install \"sudo\". \n"
+		exit 1
+	fi
+	$_SUDO -v
+	if [ $? -ne 0 ]; then
+		printf "The installer was unable to elevate privileges using \"sudo\". \n"
+		printf "Please make sure your account has \"sudo\" privileges. \n"
+	fi
+	# _ELEVATE is separate from _SUDO since environment variables may need
+	# to be passed
+	_ELEVATE="$_SUDO"
 fi
 
-# Make sure to source the latest .bashrc
-# Hack the PS1 variable to get around ubuntu .bashrc
-_OLD_PS1=$PS1
-PS1=" "
-# Hack the interactive flag to get around other .bashrc's
-set -i
-# Make sure weirdness doesn't happen with autocomplete/ etc
-set -o posix
+# ENTRYPOINT
+__entry() {
+	_UNINSTALL_RITA=false
+
+	# Optional Dependencies
+	_INSTALL_BRO=true
+	_INSTALL_MONGO=true
 
-source $HOME/.bashrc
+	# Install locations
+	_INSTALL_PREFIX=/usr/local
+	_CONFIG_PATH=/etc/rita
+	_VAR_PATH=/var/lib/rita
 
-# Clean up our hacks
-set +o posix
-set +i
-PS1=$_OLD_PS1
-unset _OLD_PS1
+	# FOR an OPT style installation
+	# NOTE: RITA itself must be changed to agree with the
+	# _CONFIG_PATH and _VAR_PATH
+	# _INSTALL_PREFIX=/opt/rita
+	# _CONFIG_PATH=/etc/opt/rita
+	# _VAR_PATH=/var/opt/rita
 
-__help() {
-	__title
+	# Parse through command args
+	while [[ $# -gt 0 ]]; do
+		case $1 in
+			-h|--help)
+				# Display help and exit
+				__help
+				exit 0
+				;;
+			-u|--uninstall)
+				_UNINSTALL_RITA=true
+				_INSTALL_BRO=false
+				_INSTALL_MONGO=false
+				;;
+			--disable-bro)
+				_INSTALL_BRO=false
+				;;
+			--disable-mongo)
+				_INSTALL_MONGO=false
+				;;
+			--prefix)
+				shift
+				_INSTALL_PREFIX="$1"
+				;;
+			--version)
+				shift
+				_RITA_VERSION="$1"
+				;;
+			*)
+			;;
+	  esac
+	  shift
+	done
 
-	cat <<HEREDOC
-Usage:
-	${_NAME} [<arguments>]
+	_BIN_PATH="$_INSTALL_PREFIX/bin"
+
+	if __installation_exist; then
+		printf "$_IMPORTANT RITA is already installed.\n"
+		printf "$_QUESTION Would you like to erase it and re-install? [y/N] "
+		read
+		if [[ $REPLY =~ ^[Yy]$ ]]; then
+			_UNINSTALL_RITA=true
+			_INSTALL_BRO=false
+			_INSTALL_MONGO=false
+		else
+			exit 0
+		fi
+	fi
 
-Options:
-	-h --help			Show this help message.
-	-u --uninstall			Remove RITA.
+	if [ "$_UNINSTALL_RITA" = "true" ]; then
+		__uninstall
+	else
+		__install
+	fi
+}
 
-HEREDOC
+__installation_exist() {
+	[ -f "$_BIN_PATH/rita" -o -d "$_CONFIG_PATH" -o -d "$_VAR_PATH" ]
 }
 
-__explain() {
-	cat <<HEREDOC
-This script will:
+__uninstall() {
+	printf "$_IMPORTANT Uninstalling RITA..."
+	if [ -f "$_BIN_PATH/rita"]; then
+		printf "$_SUBITEM Removing $_BIN_PATH/rita \n"
+		$_ELEVATE rm "$_BIN_PATH/rita"
+	else
+		printf "$_SUBIMPORTANT $_BIN_PATH/rita not found! \n"
+	fi
+	if [ -d "$_CONFIG_PATH" ]; then
+		printf "$_SUBITEM Removing $_CONFIG_PATH \n"
+		$_ELEVATE rm -rf "$_CONFIG_PATH"
```

Comment:
For uninstalling it makes sense to remove everything, including the config file. But most of our uninstall use case comes when rita is upgraded.  In that case, I'd really like to avoid wiping out the user's config file and making them start a new one.

To address this issue before, I got rid of the idea of "uninstall" and instead called it "reinstall".  This let me just overwrite files and back them up as needed instead of removing them altogether. Thoughts on doing something like this?
https://github.com/ethack/rita/blob/b651f605bf9cda083d1dcb7b8b2068f33f34cc80/install.sh#L326-L331

I really don't think the standalone uninstall is that useful if we leave behind Go, Mongo, and Bro and I don't think we should try to handle uninstalling those.  Also, the uninstall doesn't work between versions of rita where we've changed the file locations and I don't think we should try to keep track of where all the possible locations could be either to have a proper uninstall.  In practice, I haven't found people very receptive to the instruction of "use the uninstaller from the version of rita you have installed and then download the new version to run the install".  They just jump straight to running the new rita installer, which doesn't uninstall old incompatible versions.